mailbox/api/src/main/java/org/apache/james/mailbox/MailboxListener.java

```diff
@@ -63,10 +64,12 @@
     abstract class Event implements Serializable {
         private final MailboxSession session;
         private final MailboxPath path;
+        private final MailboxId mailboxId;
 
-        public Event(MailboxSession session, MailboxPath path) {
+        public Event(MailboxSession session, MailboxPath path, MailboxId mailboxId) {
```

Comment:
it's duplicate information, shouldn't we choose either MailboxPath or MailboxId ?

Reply:
If we choose, than I'm in for MailboxId

Reply:
The rationnals is that an event should not identify a resource by something "mutable".

Reply:
In some cases, mailboxPath is enough
I've tried to remove mailboxPath but there are a lot of side effect and there are some business I don't know how to handle, I would not include that so big refactor on this PR
So let's make it as deprecated, is that ok?

Reply:
No need to deprecate it. Matthieu, could you live with mailboxId and mailboxPath or should we be defensive?

Reply:
- As suspected Maildir is briken when it cames to mailbox ID.

Upon construction, we should load mailboxes into cache (for mailboxes already present: list() + cache mailbox)

All the other mailboxes implementation are fine ;-)

 - Regarding mailboxPath uses in event system:
     - GlobalRegistration (for distributed event delivery) should be done on mailboxId.
     - CacheInvalidatingMailboxListener : a mailboxId -> path lookup is needed.  No test, no use about this class. Strategy: require a MailboxMapperFactory for the lookup.
    - Distributed event serialization / deserialization need to handle mailboxIdFactory
    - ListeningCurrentQuotaUpdater needs a look-up. We need to rely on mailboxMapper for it. (Spring... )
    - **MailboxListenerSupport** needs to be implemented based on MailboxId, I guess.... Which means handling distributed registration on MailboxId and not mailboxPath